terraform-mode.el

```diff
@@ -85,6 +86,37 @@
       (add-hook 'before-save-hook #'terraform-format-buffer nil t)
     (remove-hook 'before-save-hook #'terraform-format-buffer t)))
 
+(defun interpolate (list item)
+  "Places item between elements in list"
+  (cond ((null list) nil)
+        ((= (length list) 1) list)
+        (t (append (list (car list) item)
+                   (interpolate (cdr list) item)))))
+
+(defun provider (resource-name)
+  "Returns the provider associated with a resource-name"
+  (car (split-string resource-name "_")))
+
+(defun resource (resource-name)
+  "Returns the resource associated with a resource-name"
+  (apply #'concat
+         (interpolate (cdr (split-string resource-name "_")) "_")))
+
+(defun terraform-resource-url (resource)
+  "Returns the url containing the documentation for resource"
+  (format "https://www.terraform.io/docs/providers/%s/r/%s.html"
```

Comment:
This URL format does a 301 redirect to a new URL format.  Ideally this should use the new URL format.

Reply:
Latest URLs format includes the namespace of the provider (e.g. `hashiscorp`). For instance:
```
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_cache_policy
```

This information can be retrieved in `terraform` block, which may be in another `tf` file in the same directory. Here and example:
```terraform
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 4.8"
    }
  }
}
```

I'm an elisp noob, so I don't know how to get this information ...